File: CS/WpfApplication25/ViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.Mvvm;

namespace WpfApplication25 {
    public class ViewModel {
        public virtual string Text { get; set; }
        public virtual IMessageBoxService MessageBoxService { get { return null; } }
        public void ShowText() {
            MessageBoxService.ShowMessage(Text);
        }
    }
}

File: VB/WpfApplication25/ViewModel.vb
Imports DevExpress.Mvvm

Namespace WpfApplication25

    Public Class ViewModel

        Public Overridable Property Text As String

        Public Overridable ReadOnly Property MessageBoxService As IMessageBoxService
            Get
                Return Nothing
            End Get
        End Property

        Public Sub ShowText()
            MessageBoxService.ShowMessage(Text)
        End Sub
    End Class
End Namespace
